### PR TITLE
task4600 create best answer

### PR DESCRIPTION
**Pull request by laam55**
https://edu-redmine.framgia.vn/issues/4600

### CODE REVIEW

File: app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -179,6 +179,28 @@ public function postVote(Request $request)
...
     }
 
+    public function postBestAnswer(Request $request)
+    {
+        $input = $request->all();
+
+        if ($request->ajax() && !empty($input['answer_id']) && !empty($input['post_id']) && Auth::check()) {
+            $input['user_id'] = Auth::id();
+
+            if ($result = $this->postRepository->voteBestAnswer($input)) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/PostController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.ControlStructureSpacing.SpacingAfterOpen</br>**Severity**: error</br>**File**: app/Http/Controllers/PostController.php</br>**Line**: 189</br>Blank line found at start of control structure

File: app/Repositories/Eloquents/PostRepository.php [hunk 2]
```diff
@@ -239,4 +240,30 @@ public function updateVoteTotal($id, $score)
 
         return $temp;
     }
+
+    public function voteBestAnswer($input)
+    {
+        $answer = Answer::where([
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 246</br>Expected 0 spaces before closing bracket; newline found

File: app/Repositories/Eloquents/PostRepository.php [hunk 3]
```diff
@@ -239,4 +240,30 @@ public function updateVoteTotal($id, $score)
...
+            'post_id' => $input['post_id'],
+            'id' => $input['answer_id']]
```
Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoComma</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 248</br>Each line in an array declaration must end in a comma

Comment (framgiaciv3) [inline app/Repositories/Eloquents/PostRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CloseBraceNewLine</br>**Severity**: error</br>**File**: app/Repositories/Eloquents/PostRepository.php</br>**Line**: 248</br>Closing bracket of array declaration must be on a new line

File: app/Repositories/Eloquents/PostRepository.php [hunk 4]
```diff
@@ -239,4 +240,30 @@ public function updateVoteTotal($id, $score)
...
+        ])->first();
+
+        if (!empty($answer)) {
+            $post = $this->model()->findOrFail($input['post_id']);
+
+            if ($post->user_id == $input['user_id']) {
+                if ($post->best_answer_id == $input['answer_id']) {
+                    $post->update(['best_answer_id' => 0]);
+
+                    return 'unvoted';
```
Comment (BunnyPi04) [inline app/Repositories/Eloquents/PostRepository.php]: sao lại return string?
return code đưa vào config nhé

Comment (laam55) [reply app/Repositories/Eloquents/PostRepository.php]: ok cậu

**Pull request merged**